src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -886,6 +891,15 @@ maxLength switch
                 .WithFlags(FunctionFlags.GenerateIntermediateVariableAlways)
                 .Build();
 
+            yield return new FunctionOverloadBuilder("loadYamlContent")
+               .WithGenericDescription($"Loads the specified YAML file as bicep object. File loading occurs during compilation, not at runtime.")
+               .WithRequiredParameter("filePath", LanguageConstants.StringYamlFilePath, "The path to the file that will be loaded.")
+               .WithOptionalParameter("jsonPath", LanguageConstants.String, "JSONPath expression to narrow down the loaded file. If not provided, a root element indicator '$' is used")
```

Comment:
jsonPath argument in loadYamlContent seems odd.. perhaps we can think of some other name for the argument?

Reply:
I'm open to suggestions, but the reason we stuck with 'jsonPath', is that this is not a native feature of YAML, but something we are able to provide because internally we quickly convert the YAML object into a jToken.

By keeping the name 'jsonPath', users can search that term and find more details about how to use the feature. But if we name it something like 'yamlPath', this would not return any search results. 

We considered not including this parameter, but it was so easy to enable it felt like it was worth included.

Reply:
I think this is worth more of a discussion - I agree with both of you 😄

I think what makes this conceptually confusing is that the user is going to be thinking of loading YAML "into Bicep" rather than "as JSON".

To enumerate the above (+ a few other) possibilities:
- Drop the parameter entirely. Not a huge fan of this because it was requested for JSON, and we're most likely going to want to add it back if someone asks for it.
- Rename it to something more generic like `pathFilter` and explain in the description that this adheres to the JSONPath spec. I see @dciborow's point here about syntax discoverability.
- Rename the function to something like "loadYamlAsJson" (open to better suggestions) that gives the user more of an understanding as to why a jsonPath makes sense.
- (more "out-there") Use a lambda as the 2nd argument - e.g. `loadYamlContent("path/to.yaml", root => root.foo[0].bar)`.

Reply:
I like `pathFilter`